libs/beacon/src/beacon_manager.cpp

```diff
@@ -552,9 +542,9 @@ BeaconManager::AddResult BeaconManager::AddSignaturePart(Identity const & from,
                                                          Signature const &signature)
 {
   auto it = identity_to_index_.find(from.identifier());
-  assert(it != identity_to_index_.end());
+  auto iq = qual_.find(from.identifier());
```

Comment:
iq?

Reply:
We should check that the sender is both in the cabinet and in qual.